core/shared/platform/windows/win_clock.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2023 Amazon Inc.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "platform_api_vmcore.h"
+#include <winternl.h>
+
+#define NANOSECONDS_PER_SECOND 1000000000ULL
+#define NANOSECONDS_PER_TICK 100
+
+static int
+calculate_monotonic_clock_frequency(uint64 *out_frequency)
+{
+    LARGE_INTEGER frequency;
+    if (!QueryPerformanceFrequency(&frequency)) {
+        return BHT_ERROR;
+    }
+    else {
+        *out_frequency = (uint64)frequency.QuadPart;
+        return BHT_OK;
+    }
+}
+
+// The implementation below dervies from the following source:
+// https://github.com/WasmEdge/WasmEdge/blob/b70f48c42922ce5ee7730054b6ac0b1615176285/lib/host/wasi/win.h#L210
+static uint64
+from_file_time_to_wasi_timestamp(FILETIME filetime)
+{
+    static const uint64 ntto_unix_epoch =
+        134774ULL * 86400ULL * NANOSECONDS_PER_SECOND;
+
+    ULARGE_INTEGER temp = { .LowPart = filetime.dwLowDateTime,
+                            .HighPart = filetime.dwHighDateTime };
+
+    return temp.QuadPart - ntto_unix_epoch;
+}
+
+int
+get_performance_counter_value(uint64 *out_counter)
+{
+    LARGE_INTEGER counter;
+    if (!QueryPerformanceCounter(&counter)) {
+        return BHT_ERROR;
+    }
+    else {
+        *out_counter = counter.QuadPart;
+        return BHT_OK;
+    }
+}
+
+int
+os_clock_res_get(bh_clock_id_t clock_id, uint64 *resolution)
+{
+    switch (clock_id) {
+        case BH_CLOCK_ID_MONOTONIC:
+        {
+            uint64 frequency;
+            if (calculate_monotonic_clock_frequency(&frequency) == BHT_ERROR) {
+
```

Comment:
In this file, there are blank lines after some of the if statements and before some of the 'case' blocks in the switch statement. I think we can remove them.